Pass additional plot arguments to `DiagrammeR::grViz()` and fix #2

lewinfox opened the pull request:
See #2, #3. 

@SigurdJanson I don't seem to see any difference in behaviour when passing additional args to `plot()`. Would you be able to checkout the `develop` branch and try out the new code?

SigurdJanson committed:
Used the ellipsis argument to actually pass additional arguments to the plotting function.

lewinfox committed:
Update README and add NEWS.md

SigurdJanson commented:
I wish I could. I tried using the examples from the readme. I always get errors which can be tracked down to `get_calls()`.

Try this test:

SigurdJanson commented:
I can't. I get this error when I use the example taken from the readme.

> Error in ul[!sapply(ul, rlang::is_missing)] : 
> object of type 'symbol' is not subsettable

lewinfox commented:
Yeah, same here. A bug to be squashed! Thanks for your help, I’ll se what I
can do.

On Sun, 7 Aug 2022 at 5:48 AM, SigurdJanson ***@***.***>
wrote:

> I can't. I get this error when I use the example taken from the readme.
>
> Error in ul[!sapply(ul, rlang::is_missing)] :
> object of type 'symbol' is not subsettable
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lewinfox/foodwebr/pull/4#issuecomment-1207254509>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGC6L2SKBEPQKA7WJAGO7BLVX2QO3ANCNFSM55XZOEXQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

SigurdJanson commented:
Maybe I can help. The test below nails it. The problem lies in `get_calls`. When I replace line 32 with `return(list(expr[[1L]]))` it works. The problem is that the caller `get_called_functions` tries to subset the result of `get_calls`. But in line 32 `get_calls` returns a `symbol`.

But be careful. I do not know if I cause any unwanted side effects. Wit this solution I get numerous other errors.

```r
bb <- function() seq_len(3)
test_that("`get_calls(bb)`: with `bb` being a single call to other function without brackets", {
  # act
  cls <- get_calls(body(bb))
  cls <- unlist(cls)

  # assert
  expect_identical(cls[[1]], quote(seq_len))
})
```

lewinfox committed:
Remove dev dependencies and `renv::init()`

We will use `renv` to decide what development packages to install rather
than using `Suggests`.

lewinfox committed:
Merge branch 'develop' into feature/renv

lewinfox committed:
Merge pull request #7 from lewinfox/feature/renv

Add `renv`

lewinfox committed:
Change the detection of function calls

Using `codetools` simplifies the process of identifying function calls.
Fixes #2.

lewinfox committed:
Remove redundant `tokenise_function()` function

lewinfox committed:
Unskip test

lewinfox committed:
Update docs and DESCRIPTION

lewinfox committed:
Add codetools to renv

lewinfox committed:
Fix the case for cowsay::say

lewinfox committed:
Revert to using `lsf.str()`

lewinfox committed:
Update README

lewinfox committed:
Update comments

lewinfox committed:
Merge pull request #8 from lewinfox/bug/fix-fn-detection

Fix function detection

lewinfox committed:
Bump version to 1.0.0

lewinfox committed:
Fix NEWS.md as there was no intermediate release

lewinfox commented:
@SigurdJanson sorry for the very long delay merging this - for various reasons this has been way down my priority list. Your changes are now in `main` and hopefully the package will be released to CRAN in the next week or so. Thanks again for your contribution!
